addons/web/static/src/core/utils/numbers.js

```diff
@@ -156,3 +156,43 @@ export function humanNumber(number, options = { decimals: 0, minDigits: 1 }) {
     }
     return int + decimalPoint + decimalPart + symbol;
 }
+
+/**
+ * Returns a string representing a float.  The result takes into account the
+ * user settings (to display the correct decimal separator).
+ *
+ * @param {number} value the value that should be formatted
+ * @param {Object} [options]
+ * @param {number[]} [options.digits] the number of digits that should be used,
+ *   instead of the default digits precision in the field.
+ * @param {boolean} [options.humanReadable] if true, large numbers are formatted
+ *   to a human readable format.
+ * @param {string} [options.decimalPoint] decimal separating character
+ * @param {string} [options.thousandsSep] thousands separator to insert
+ * @param {number[]} [options.grouping] array of relative offsets at which to
+ *   insert `thousandsSep`. See `insertThousandsSep` method.
+ * @param {number} [options.decimals] used for humanNumber formmatter
+ * @param {boolean} [options.noTrailingZeros=false] if true, the decimal part
+ *   won't contain unnecessary trailing zeros.
+ * @returns {string}
+ */
+export function formatFloat(value, options = {}) {
+    if (options.humanReadable) {
+        return humanNumber(value, options);
+    }
+    const grouping = options.grouping || l10n.grouping;
+    const thousandsSep = "thousandsSep" in options ? options.thousandsSep : l10n.thousandsSep;
+    const decimalPoint = "decimalPoint" in options ? options.decimalPoint : l10n.decimalPoint;
+    let precision;
+    if (options.digits && options.digits[1] !== undefined) {
+        precision = options.digits[1];
+    } else {
+        precision = 2;
+    }
+    const formatted = value.toFixed(precision).split(".");
+    formatted[0] = insertThousandsSep(formatted[0], thousandsSep, grouping);
+    if (options.noTrailingZeros) {
```

Comment:
xmo is right, this would crash if precision is 0
```suggestion
    if (options.noTrailingZeros && formatted[1]) {
```